pytest_rerunfailures.py

```diff
@@ -106,6 +115,86 @@ class RerunPlugin(object):
 
     def __init__(self):
         self.tests_to_rerun = set([])
+        self.testrun_stats = {}
+        self.rerun_stats = {
+            'rerun_tests': [],
+            'total_failed': 0,
+            'total_reruns': 0,
+            'total_resolved_by_reruns': 0
+        }
+
+    def _failed_test(self, nodeid):
+        """
+        Template format for failed test stat
+
+        Returns
+        -------
+        dict
+        """
+        stat = self.testrun_stats.get(nodeid, {
+            'nodeid': nodeid,
+            'status': 'failed',
+            'setup': {
+                'caplog': None,
+                'capstderr': None,
+                'capstdout': None,
+                'text_repr': None
+            },
+            'call': {
+                'caplog': None,
+                'capstderr': None,
+                'capstdout': None,
+                'text_repr': None
+            },
+            'teardown': {
+                'caplog': None,
+                'capstderr': None,
+                'capstdout': None,
+                'text_repr': None
+            },
```

Comment:
Why don't we have `original_trace` here?

Reply:
I guess it's first failure, which is original as opposite to rerun

Reply:
@andrew-qa Is right - `_failed_test` is used for `self.testrun_stats` and we interested only in failed tests. So no possible original trace here, since it is original itself. As opposite `_rerun_test` is used for `self.rerun_stats` and will have a reference to original trace.